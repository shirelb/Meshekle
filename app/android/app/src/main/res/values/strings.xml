<resources>
    <string name="app_name">משקלה</string>
    <string name="default_notification_channel_id" translatable="false">fcm_default_channel</string>
</resources>
