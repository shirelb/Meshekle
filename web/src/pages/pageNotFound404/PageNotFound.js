import React from 'react';
import {Grid, Header, Image} from 'semantic-ui-react';

const PageNotFound = () => (
    <Grid stretched textAlign={"center"}>
        <Grid.Column stretched textAlign={"center"}>
            <Image centered src={require('../../images/logo1.png')} size={'large'}/>
            <Header size={"huge"} textAlign={"center"} icon style={{marginTop: "-10px"}}>
                משקל'ה
                <Header.Subheader>יש משק ויש משקל'ה</Header.Subheader>
            </Header>

            <div style={{marginTop: 20, textAlign: "center"}}>
                <Header as={'h1'}> 404 </Header>
                <Header as={'h2'}> Page not found </Header>
            </div>
        </Grid.Column>
    </Grid>
);

export default PageNotFound;