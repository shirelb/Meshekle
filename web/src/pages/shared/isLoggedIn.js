import store from 'store';
import axios from 'axios';
import {SERVER_URL} from "../shared/constants";

export default () => {
    // !!store.get('loggedIn') ?
    var serviceProviderToken = store.get('serviceProviderToken');
    console.log("serviceProviderToken ", serviceProviderToken);

    if (serviceProviderToken === null || serviceProviderToken === undefined)
        return false;
    else {
        axios.post(`${SERVER_URL}/api/serviceProviders/validToken`,
            {
                "token": serviceProviderToken,
            },
        )
            .then(validTokenResponse => {
                console.log("validTokenResponse ", validTokenResponse);
                store.set('serviceProviderId', validTokenResponse.data.payload.serviceProviderId);
                store.set('userId', validTokenResponse.data.payload.userId);
            })
            .catch((error) => {
                console.log(error);
            });
        return true;
    }
}