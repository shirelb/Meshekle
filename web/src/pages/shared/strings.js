const loginPageStrings = {
    LOGIN: "התחבר",
    SUBMIT: "שלח",
    WELCOME_TO_LOGIN: "Welcome to the login screen!",
    USER_ID_PLACEHOLDER: "ת.ז.",
    PASSWORD_PLACEHOLDER: "סיסמא",
    APP_NAME: "משקל\'ה",
    WRONG_CREDENTIALS: "אחד הפרטים שהזנת לא נכון",
    NO_SUCH_USER: "המשתמש אינו קיים",
    EMPTY_USER_ID: "הוסף ת.ז. ונסה להתחבר שוב",
    EMPTY_PASSWORD: "הוסף סיסמא ונסה להתחבר שוב",
};

const mainScreenStrings = {
    LOGOUT: "התנתק",
    WELCOME: "ברוכים הבאים ",
};

const drawerMenu = {
    LOGOUT: "התנתק",
    MainScreenName: "בית",
    AppointmentsScreenName: "תורים",
    ChoresScreenName: "תורנויות",
};

export default {loginPageStrings, mainScreenStrings, drawerMenu};