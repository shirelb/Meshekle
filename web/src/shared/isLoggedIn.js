import store from 'store';
import axios from 'axios';
import {SERVER_URL} from "./constants";

export default () => {
    // !!store.get('loggedIn') ?
    var serviceProviderToken = store.get('serviceProviderToken');
    console.log("serviceProviderToken ", serviceProviderToken);

    if (serviceProviderToken === null || serviceProviderToken === undefined)
        return new Promise(resolve => false);
    else {
        return axios.post(`${SERVER_URL}/api/serviceProviders/validToken`,
            {
                "token": serviceProviderToken,
            },
        )
            .then(validTokenResponse => {
                console.log("validTokenResponse ", validTokenResponse);
                store.set('serviceProviderId', validTokenResponse.data.payload.serviceProviderId);
                store.set('userId', validTokenResponse.data.payload.userId);
                return true;
            })
            .catch((error) => {
                console.log(error);
                return false;
            });
        // return true;
        // setTimeout(() => {
        // }, 1000)
    }
}