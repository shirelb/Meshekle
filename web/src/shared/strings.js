const loginPageStrings = {
    LOGIN: "התחבר",
    SUBMIT: "שלח",
    WELCOME_TO_LOGIN: "Welcome to the login screen!",
    USER_ID_PLACEHOLDER: "ת.ז.",
    PASSWORD_PLACEHOLDER: "סיסמא",
    APP_NAME: "משקל\'ה",
    WRONG_CREDENTIALS: "אחד הפרטים שהזנת לא נכון",
    NO_SUCH_USER: "המשתמש אינו קיים",
    EMPTY_USER_ID: "הוסף ת.ז. ונסה להתחבר שוב",
    EMPTY_PASSWORD: "הוסף סיסמא ונסה להתחבר שוב",
};

const mainPageStrings = {
    LOGOUT: "התנתק",
    WELCOME: "ברוכים הבאים ",
    MAIN_PAGE_TITLE: "בית",
    SETTINGS_PAGE_TITLE: "הגדרות",
    PHONE_BOOK_PAGE_TITLE: "ספר טלפונים",
    PHONE_BOOK_PAGE_USERS_TITLE: "משתמשים",
    PHONE_BOOK_PAGE_SERVICE_PROVIDERS_TITLE: "נותני שירות",
    APPOINTMENTS_PAGE_TITLE: "תורים",
    CHORES_PAGE_TITLE: "תורנויות",
    ANNOUNCEMENTS_PAGE_TITLE: "לוח מודעות",
    ROLES_HairDresser: "מספרה",
    ROLES_Admin: "מנהלה",
    ROLES_Dentist: "רפואת שיניים",
    ROLES_Undefined: "לא נמצאו תפקידים מתאימים",

};

const phoneBookPageStrings = {
    ADD_SERVICE_PROVIDER: "נותן שירות חדש",
    ADD_USER: "משתמש חדש",
    EDIT_USER: "ערוך משתמש",
    USER_ID_HEADER: "ת.ז.",
    FULLNAME_HEADER: "שם מלא",
    PASSWORD_HEADER: "סיסמא",
    EMAIL_HEADER: "אימייל",
    MAILBOX_HEADER: "תיבת דואר",
    CELLPHONE_HEADER: "פלאפון",
    PHONE_HEADER: "טלפון",
    BORN_DATE_HEADER: "תאריך לידה",
    ACTIVE_HEADER: "קיים",
    ACTIVE_ANSWER_YES: "כן",
    ACTIVE_ANSWER_NO: "לא",
    SERVICE_PROVIDER_ID_HEADER: "ת.ז. נותן שירות",
    SERVICE_PROVIDER_ROLE_HEADER: "תפקיד",
    SERVICE_PROVIDER_USER_ID_HEADER: "ת.ז. יוזר",
    SERVICE_PROVIDER_OPERATION_TIME_HEADER: "זמני פעילות",
    SERVICE_PROVIDER_APPOINTMENT_WAY_TYPE_HEADER: "דרך קביעת תורים",
};

const appointmentsPageStrings = {
    ADD_APPOINTMENT: "תור חדש",
    APPOINTMENT_ID: "מספר מזהה",
    CLIENT_NAME: "שם משתמש",
    CLIENT_ID: "ת.ז. משתמש",
    SERVICE_PROVIDER_ID: "ת.ז. נותן שירות",
    ROLE: "ענף",
    SUBJECT: "נושא",
    STATUS: "סטאטוס",
    DATE: "תאריך",
    START_TIME: "זמן התחלה",
    END_TIME: "זמן סיום",
    REMARKS: "הערות",
    OPTIONAL_TIMES: "זמנים אופציונאליים",
};

const announcementsPageStrings = {
    ANNOUNCE_REQ_TITLE_TABLE: "מודעות המחכות לאישור",
    ANNOUNCE_TITLE_TABLE: "מודעות באחריותך",
    ANNOUNCE_NUMBER:"מספר מודעה",
    ANNOUNCE_USERNAME:"שם המפרסם",
    ANNOUNCE_CATEGORY:"קטגוריה",
    ANNOUNCE_TITLE:"נושא",
    ANNOUNCE_CONTENT:"תוכן",
    ANNOUNCE_STATUS:"סטטוס",
    ANNOUNCE_FILE:"קובץ מצורף",
    ANNOUNCE_DATE_OF_EVENT:"תאריך אירוע",
    ANNOUNCE_EXPR_DATE:"תאריך תפוגה",
    OPERATION_OPTIONS:"אפשרויות",
    ADD_ANNOUNCEMENT:"פרסם מודעה חדשה",
};

const roles = {
    HairDresser: "מספרה",
    Admin: "מנהלה",
    Dentist: "רפואת שיניים",
};


const appointmentsWayType = {
    Dialog: "דיון",
    Slots: "שורה נעלמת",
    Fault: "פתיחת תקלה",
    Admin: "הכל",
};

export default {
    loginPageStrings,
    mainPageStrings,
    phoneBookPageStrings,
    appointmentsPageStrings,
    roles,
    appointmentsWayType,
    announcementsPageStrings,
};
