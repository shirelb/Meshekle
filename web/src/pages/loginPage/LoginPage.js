import React, {Component} from 'react';
import './styles.css'
import 'semantic-ui-css/semantic.min.css';
import {Form, Grid, Header, Message} from 'semantic-ui-react';
import {Helmet} from 'react-helmet';
import store from 'store';
import axios from 'axios';
import {Redirect} from 'react-router-dom';
import isLoggedIn from '../shared/isLoggedIn';
import {SERVER_URL} from "../shared/constants";
import strings from '../shared/strings'
import mappers from '../shared/mappers'

class LoginPage extends Component {

    constructor(props) {
        super(props);

        this.state = {
            userId: '',
            password: '',
            error: false,
            err: []
        };

        this.handleChange = this.handleChange.bind(this);
        this.onSubmit = this.onSubmit.bind(this);

        // isLoggedIn(props);
        /*this.isLoggedIn = false;

        if (isLoggedIn()) {
            this.isLoggedIn = true;
            console.log("componentDidMount isLoggedIn === true");
        }*/
    }

    /* componentDidMount() {
         console.log("in componentDidMount");
         if (isLoggedIn()) {
             this.isLoggedIn = true;
             console.log("componentDidMount isLoggedIn === true");
         }
         console.log("componentDidMount isLoggedIn === false");
     }*/

    validate = (userId, password) => {
        console.log('in validate func');
        const errors = [];
        let item = strings.loginPageStrings.WRONG_CREDENTIALS;

        if (userId.length === 0) {
            errors.push(strings.loginPageStrings.EMPTY_USER_ID);
        } else {

            if (userId.length > 9) {
                if (errors.indexOf(item) === -1) errors.push(item);
            }
            if (!(/^\d+$/.test(userId))) {
                if (errors.indexOf(item) === -1) errors.push(item);
            }
        }


        if (password.length === 0) {
            errors.push(strings.loginPageStrings.EMPTY_PASSWORD);
        } else {
            if (password.length < 8) {
                if (errors.indexOf(item) === -1) errors.push(item);
            }
            if (password.length > 12) {
                if (errors.indexOf(item) === -1) errors.push(item);
            }
            if (!(/\d/.test(password) && /[a-zA-Z]/.test(password))) {
                if (errors.indexOf(item) === -1) errors.push(item);
            }
        }

        return errors;
    };

    onSubmit(e) {
        e.preventDefault();

        const {userId, password} = this.state;
        const { history } = this.props;

        this.setState({error: false});

        const errors = this.validate(userId, password);
        if (errors.length > 0) {
            console.log(errors);
            this.setState({error: true});
            this.setState({err: errors});
        } else {
            axios.post(`${SERVER_URL}/api/serviceProviders/login/authenticate`,
                {
                    "userId": this.state.userId,
                    "password": this.state.password
                },
            )
                .then((response) => {
                    console.log(response);
                    store.set('serviceProviderToken', response.data.token);
                    axios.post(`${SERVER_URL}/api/serviceProviders/validToken`,
                        {
                            "token": response.data.token
                        },
                    )
                        .then((validTokenResponse) => {
                            console.log(validTokenResponse);
                            store.set('serviceProviderId', validTokenResponse.data.payload.serviceProviderId);
                            store.set('userId', validTokenResponse.data.payload.userId);
                            console.log("you're logged in. yay!");
                            // store.set('loggedIn', true);
                            // this.isLoggedIn = true;
                            // this.props.history.push("/");
                            history.push('/users');
                            // this.forceUpdate();
                        });
                })

                .catch((error) => {
                    let msg = mappers.loginPageMapper(error.response.data.message);
                    this.setState({err: [msg]});
                    this.setState({error: true});

                    // console.log('in auth msg: ',msg);
                    // if (this.state.err.indexOf(msg) === -1) this.setState({err: [msg]});
                });

            /*if (!(username === 'george' && password === 'foreman')) {
                return this.setState({ error: true });
            }*/

// console.log("you're logged in. yay!");
// store.set('loggedIn', true);
        }
    }

    handleChange(e, {name, value}) {
        this.setState({[name]: value});
    }

    render() {
        if (isLoggedIn()) {
            return <Redirect to="/users" />;
        }
        /*if (isLoggedIn()) {
            console.log("isLoggedIn === true");
            return <Redirect to="/users"/>;
        }*/
        // console.log("isLoggedIn === true");
        // return <Redirect to="/users"/>;
        console.log("isLoggedIn === false");

        const {error} = this.state;

        return (
            <Grid>
                <Helmet>
                    <title>CMS | Login</title>
                </Helmet>

                <Grid.Column width={6}/>
                <Grid.Column width={4}>
                    <Form className="loginForm" error={error} onSubmit={this.onSubmit}>
                        <Header as="h1">{strings.loginPageStrings.LOGIN}</Header>
                        {error && <Message
                            error={error}
                            // content="That username/password is incorrect. Try again!"
                            content={strings.loginPageStrings.WRONG_CREDENTIALS}
                        />}
                        <Form.Input
                            inline
                            // label="Username"
                            label={strings.loginPageStrings.USER_ID_PLACEHOLDER}
                            name="userId"
                            onChange={this.handleChange}
                        />
                        <Form.Input
                            inline
                            // label="Password"
                            label={strings.loginPageStrings.PASSWORD_PLACEHOLDER}
                            type="password"
                            name="password"
                            onChange={this.handleChange}
                        />
                        <Form.Button type="submit">{strings.loginPageStrings.SUBMIT}</Form.Button>
                    </Form>
                </Grid.Column>
            </Grid>
        );
    }
}

/*
class LoginPage extends Component {

    // Using a class based component here because we're accessing DOM refs

    handleSignIn(e) {
        e.preventDefault();
        let username = this.refs.username.value;
        let password = this.refs.password.value;

        axios.get("http://localhost:4000/users")
            .then(response => {
                console.log("after get response");
                console.log(response);
                this.props.onSignIn(username, password)
            })
            .catch(error => {
                console.log("after get error");
                console.log(error);
            });
    }

    onSubmit(e) {
        e.preventDefault();

        const {username, password} = this.state;
        const {history} = this.props;

        this.setState({error: false});

        if (!(username === 'george' && password === 'foreman')) {
            return this.setState({error: true});
        }

        store.set('loggedIn', true);
        history.push('/users');
    }

    render() {
        return (
            <form onSubmit={this.handleSignIn.bind(this)}>
                <h3>Sign in</h3>
                <input type="text" ref="username" placeholder="enter you username" />
                <input type="password" ref="password" placeholder="enter password" />
                <input type="submit" value="Login" />
            </form>
        )
    }

}
*/

export default LoginPage;
