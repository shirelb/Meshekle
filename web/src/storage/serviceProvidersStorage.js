import axios from "axios";
import {SERVER_URL} from "../shared/constants";
import store from "store";

const serviceProviderHeaders = {
    'Authorization': 'Bearer ' + store.get('serviceProviderToken')
};

var getServiceProviders = function () {
    return axios.get(`${SERVER_URL}/api/serviceProviders`,
        {headers: serviceProviderHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get serviceProviders error ', error)
        });
};

var getRolesOfServiceProvider = (serviceProviderId) => {
    return axios.get(`${SERVER_URL}/api/serviceProviders/roles/serviceProviderId/${serviceProviderId}`,
        {headers: serviceProviderHeaders}
    )
        .then((response) => {
            let roles = response.data;
            console.log('roles ', roles);
            return roles;
        })
        .catch((error) => {
            console.log('get serviceProvider roles error ', error)
        });
};

var serviceProviderLogin = (userId, password) => {
    return axios.post(`${SERVER_URL}/api/serviceProviders/login/authenticate`,
        {
            "userId": userId,
            "password": password
        },
    );
};

var serviceProviderValidToken = function (token) {
    return axios.post(`${SERVER_URL}/api/serviceProviders/validToken`,
        {
            "token": token
        },
    );
}

var getServiceProviderPermissionsById = (serviceProviderId) => {
    return axios.get(`${SERVER_URL}/api/serviceProviders/serviceProviderId/${serviceProviderId}/permissions`,
        {headers: serviceProviderHeaders}
    )
        .then((response) => {
            let permissions = response.data;
            console.log('permissions ', permissions);
            return permissions;
        })
        .catch((error) => {
            console.log('error ', error);
        });
};


export default {
    getServiceProviders,
    getRolesOfServiceProvider,
    serviceProviderLogin,
    serviceProviderValidToken,
    getServiceProviderPermissionsById,

};
