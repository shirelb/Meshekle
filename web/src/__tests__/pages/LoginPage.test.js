import React from 'react';


import {Redirect} from 'react-router-dom';
import {mount, shallow} from "enzyme/build";
import PageNotFound from "../../pages/pageNotFound404/PageNotFound";
import serviceProvidersStorage from "../../storage/serviceProvidersStorage";
import LoginPage from "../../pages/loginPage/LoginPage";
import isLoggedIn from "../../shared/isLoggedIn";
import strings from "../../shared/strings";
import {setupComponent} from "../testHelpers";

jest.mock("../../shared/isLoggedIn");
jest.mock("../../storage/serviceProvidersStorage");


describe("LoginPage should", () => {
    let wrapper = null;
    let componentInstance = null;
    const props = {
        location: {
            pathname: '/login'
        },
        match: {
            isExact: true,
            path: '/login',
            url: '/login',
        }
    };
    const userIdTest = "549963652";
    const passwordTest = "Qw345678";

    beforeEach(() => {
        jest.clearAllMocks();
    });


    test("renders Login when user is NOT authenticated", async () => {
        isLoggedIn.mockResolvedValue(false);
        const arrResponse = await setupComponent('shallow', LoginPage, null, props, "/login");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        expect(componentInstance).toMatchSnapshot();

        expect(wrapper.find(LoginPage)).toHaveLength(1);
        expect(wrapper.find(Redirect)).toHaveLength(0);
        expect(wrapper.find(PageNotFound)).toHaveLength(0);
    });

    test("renders Redirect when user is authenticated", async () => {
        isLoggedIn.mockResolvedValue(true);
        const arrResponse = await setupComponent('mount', LoginPage, null, props, "/login");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];
        // await setupLoginPage(true, 'mount');

        expect(wrapper.find(Redirect)).toHaveLength(1);
        expect(wrapper.find('Router').prop('history').location.pathname).toEqual('/home')
    });

    test("render a page with header and form with two fields and a send button", async () => {
        isLoggedIn.mockResolvedValue(false);
        const arrResponse = await setupComponent('shallow', LoginPage, null, props, "/login");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];
        const shallowWrapper = wrapper.find('LoginPage').dive();

        expect(shallowWrapper.find('Form')).toHaveLength(1);
        expect(shallowWrapper.find('Header')).toHaveLength(1);
        expect(shallowWrapper.find('Header').dive().text()).toEqual(strings.loginPageStrings.LOGIN);
        expect(shallowWrapper.find('FormInput')).toHaveLength(2);
        expect(shallowWrapper.find('FormInput').first().props().name).toEqual('userId');
        expect(shallowWrapper.find('FormInput').first().props().label).toEqual(strings.loginPageStrings.USER_ID_PLACEHOLDER);
        expect(shallowWrapper.find('FormInput').get(1).props.name).toEqual('password');
        expect(shallowWrapper.find('FormInput').get(1).props.label).toEqual(strings.loginPageStrings.PASSWORD_PLACEHOLDER);
        expect(shallowWrapper.find('FormButton')).toHaveLength(1);
        expect(shallowWrapper.find('FormButton').props().children).toEqual(strings.loginPageStrings.SUBMIT);
        expect(shallowWrapper.find('FormButton').props().type).toEqual('submit');

        expect(componentInstance.state.err.length).toEqual(0);
        expect(componentInstance.state.err).toEqual([]);
        expect(componentInstance.state.error).toBeFalsy();
        expect(shallowWrapper.find('Message')).toHaveLength(0);
    });

    test("handle change of userId", async () => {
        const handleChangeSpy = jest.spyOn(LoginPage.prototype, 'handleChange');

        isLoggedIn.mockResolvedValue(false);
        const arrResponse = await setupComponent('shallow', LoginPage, null, props, "/login");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        expect(componentInstance.state.userId).toEqual('');

        const userIdInput = wrapper.find('LoginPage').dive().find('FormInput').first();
        await userIdInput.simulate('change', {
            target: {
                name: 'userId',
                value: userIdTest
            }
        });
        expect(handleChangeSpy).toHaveBeenCalled();
        expect(handleChangeSpy.mock.instances[0].state.userId).toEqual(userIdTest);
    });

    test("handle change of password", async () => {
        const handleChangeSpy = jest.spyOn(LoginPage.prototype, 'handleChange');

        isLoggedIn.mockResolvedValue(false);
        const arrResponse = await setupComponent('shallow', LoginPage, null, props, "/login");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        expect(componentInstance.state.password).toEqual('');

        const passwordInput = wrapper.find('LoginPage').dive().find('FormInput').at(1);
        await passwordInput.simulate('change', {
            target: {
                name: 'password',
                value: passwordTest
            }
        });
        expect(handleChangeSpy).toHaveBeenCalled();
        expect(handleChangeSpy.mock.instances[0].state.password).toEqual(passwordTest);
    });

    test("validate userId and password", async () => {
        isLoggedIn.mockResolvedValue(false);
        const arrResponse = await setupComponent('shallow', LoginPage, null, props, "/login");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        let validateResponse = componentInstance.validate("", passwordTest);
        expect(validateResponse.length).toEqual(1);
        expect(validateResponse[0]).toEqual(strings.loginPageStrings.EMPTY_USER_ID);

        validateResponse = componentInstance.validate("11111111111111111", passwordTest);
        expect(validateResponse.length).toEqual(1);
        expect(validateResponse[0]).toEqual(strings.loginPageStrings.WRONG_CREDENTIALS);

        validateResponse = componentInstance.validate("user223", passwordTest);
        expect(validateResponse.length).toEqual(1);
        expect(validateResponse[0]).toEqual(strings.loginPageStrings.WRONG_CREDENTIALS);

        validateResponse = componentInstance.validate(userIdTest, "");
        expect(validateResponse.length).toEqual(1);
        expect(validateResponse[0]).toEqual(strings.loginPageStrings.EMPTY_PASSWORD);

        validateResponse = componentInstance.validate(userIdTest, "5");
        expect(validateResponse.length).toEqual(1);
        expect(validateResponse[0]).toEqual(strings.loginPageStrings.WRONG_CREDENTIALS);

        validateResponse = componentInstance.validate(userIdTest, "55555555555555555555555555");
        expect(validateResponse.length).toEqual(1);
        expect(validateResponse[0]).toEqual(strings.loginPageStrings.WRONG_CREDENTIALS);

        validateResponse = componentInstance.validate(userIdTest, "סיסמא");
        expect(validateResponse.length).toEqual(1);
        expect(validateResponse[0]).toEqual(strings.loginPageStrings.WRONG_CREDENTIALS);

        validateResponse = componentInstance.validate("", "סיסמא");
        expect(validateResponse.length).toEqual(2);
        expect(validateResponse[0]).toEqual(strings.loginPageStrings.EMPTY_USER_ID);
        expect(validateResponse[1]).toEqual(strings.loginPageStrings.WRONG_CREDENTIALS);

        validateResponse = componentInstance.validate(userIdTest, passwordTest);
        expect(validateResponse.length).toEqual(0);
    });

    test("submit login form with wrong credentials", async () => {
        const onSubmitSpy = jest.spyOn(LoginPage.prototype, 'onSubmit');

        isLoggedIn.mockResolvedValue(false);
        const arrResponse = await setupComponent('mount', LoginPage, null, props, "/login");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        await componentInstance.setState({userId: "", password: ""});
        const validateSpy = jest.spyOn(componentInstance, 'validate');
        validateSpy.mockReturnValue([strings.loginPageStrings.EMPTY_USER_ID, strings.loginPageStrings.EMPTY_PASSWORD]);

        const loginForm = wrapper.find('LoginPage').find('Form');
        await loginForm.simulate('submit', {
            preventDefault() {
            }
        });

        expect(onSubmitSpy).toHaveBeenCalled();
        expect(componentInstance.state.error).toBeTruthy();
        expect(componentInstance.state.err).toHaveLength(2);
        expect(wrapper.find('LoginPage').find('Message')).toHaveLength(1);
        expect(wrapper.find('LoginPage').find('Message').props().content).toEqual(strings.loginPageStrings.WRONG_CREDENTIALS);
    });

    test("submit login form with right credentials", async () => {
        const onSubmitSpy = jest.spyOn(LoginPage.prototype, 'onSubmit');
        serviceProvidersStorage.serviceProviderLogin.mockResolvedValue({
            "success": true,
            "message": "Token generated successfully !",
            "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzZXJ2aWNlUHJvdmlkZXJJZCI6IjU0OTk2MzY1MiIsInVzZXJJZCI6IjU0OTk2MzY1MiIsImlhdCI6MTU1ODYyNjk3MSwiZXhwIjoxNTU4NjYyOTcxfQ.r4oH3N4qvGwRVAFMMRsH5Ls7hC1SQGDme7Gw_bipOc0"
        });

        isLoggedIn.mockResolvedValue(false);
        const arrResponse = await setupComponent('mount', LoginPage, null, props, "/login");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        await componentInstance.setState({userId: userIdTest, password: passwordTest});
        const validateSpy = jest.spyOn(componentInstance, 'validate');
        validateSpy.mockReturnValue([]);

        const loginForm = wrapper.find('LoginPage').find('Form');
        await loginForm.simulate('submit', {
            preventDefault() {
            }
        });

        expect(onSubmitSpy).toHaveBeenCalled();

        expect(componentInstance.state.err.length).toEqual(0);
        expect(componentInstance.state.err).toEqual([]);
        expect(componentInstance.state.error).toBeFalsy();
        expect(wrapper.find('LoginPage').find('Message')).toHaveLength(0);
    });
});
