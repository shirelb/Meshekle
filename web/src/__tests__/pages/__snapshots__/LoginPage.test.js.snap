// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should renders Login when user is NOT authenticated 1`] = `
LoginPage {
  "context": Object {},
  "handleChange": [Function],
  "onSubmit": [Function],
  "props": Object {
    "location": Object {
      "pathname": "/login",
    },
    "match": Object {
      "isExact": true,
      "path": "/login",
      "url": "/login",
    },
  },
  "refs": Object {},
  "setState": [Function],
  "state": Object {
    "err": Array [],
    "error": false,
    "isLoggedIn": false,
    "password": "",
    "redirectToReferrer": false,
    "userId": "",
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <LoginPage
        location={
          Object {
            "pathname": "/login",
          }
        }
        match={
          Object {
            "isExact": true,
            "path": "/login",
            "url": "/login",
          }
        }
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <Grid>
        <HelmetWrapper
          defer={true}
          encodeSpecialCharacters={true}
        >
          <title>
            Meshekle | Login
          </title>
        </HelmetWrapper>
        <GridColumn
          width={6}
        />
        <GridColumn
          width={4}
        >
          <Form
            as="form"
            className="loginForm"
            error={false}
            onSubmit={[Function]}
          >
            <Header
              as="h1"
            >
              התחבר
            </Header>
            <FormInput
              as={[Function]}
              control={[Function]}
              inline={true}
              label="ת.ז."
              name="userId"
              onChange={[Function]}
            />
            <FormInput
              as={[Function]}
              control={[Function]}
              inline={true}
              label="סיסמא"
              name="password"
              onChange={[Function]}
              type="password"
            />
            <FormButton
              as={[Function]}
              control={[Function]}
              type="submit"
            >
              שלח
            </FormButton>
          </Form>
        </GridColumn>
      </Grid>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
  "validate": [Function],
}
`;
